specifyweb/frontend/js_src/lib/components/TreeView/index.tsx

```diff
@@ -386,24 +405,59 @@ export function TreeViewWrapper(): JSX.Element | null {
       : undefined;
 
   if (treeName === undefined || !isTreeModel(treeName)) return <NotFoundView />;
+
+  const [isSplit, setIsSplit] = React.useState(false);
+
+  const treeContainer = (isFirst: boolean) => {
+    if (typeof treeDefinition === 'object') {
+      return (
+        <IsTreeSplitContext.Provider value={isSplit}>
+          <TreeView
+            treeDefinition={treeDefinition.definition}
+            treeDefinitionItems={treeDefinition.ranks}
+            tableName={treeName}
+            /**
+             * We're casting this as a generic Specify Resource because
+             * Typescript complains that the get method for each member of the
+             * union type of AnyTree is not compatible
+             *
+             */
+            key={(treeDefinition.definition as SpecifyResource<AnySchema>).get(
+              'resource_uri'
+            )}
+            isSplit={[isSplit, setIsSplit]}
+            isFirst={isFirst}
+          />
+        </IsTreeSplitContext.Provider>
+      );
+    } else {
+      return null;
+    }
+  };
+
```

Comment:
@maxxxxxdlp , 
What would be your suggestion to have a re-render of the first TreeView when doing an edit in the second and vice-versa? At the moment a page reload is necessary

Reply:
since each tree row fetches it's children, rather than having all fetched children in the `<TreeView>` component, you have some choices:
1. Refactor the code so that the "rows" state for all rows is stored in "TreeRow"
2. Set event listeners and listen for them. remember you set an event listener for queries being deleted, so that the query list dialog can remove them from the list? this would be similar but on a larger scale
    - an event is emited when tree item is deleted - all rows listen for that event and check if one of their children is deleted, and if so, update the state accordingly
    - an event is emited when tree item is created - all rows listen for that event and check if they are the parent of the new child, and if so, update the state accordingly
3. Make it so that the two TreeRows that represent the same tree item share the same "rows" state - i.e instead of `const [rows, setRows] = React.useState<RA<Row> | undefined>(undefined);`, do `const [rows, setRows] = useRows(row.nodeId);` - and that hooks checks if it was already called with a node with this id, and if so, returns the same "rows" state rather than creating a new one

My recommendation is for option 2 or 3
2 is the simplest, and would benefit other places too (setting events and listening to them would be useful for other places in specify - i.e it already benefited the query list dialog by letting it delete removed queries - it could benefit other places in the future too)
3 is the best in terms of performance - this way two split trees won't have to fetch the same children again, as they are sharing the children

feel free to ask Jason or Vinny or me for help on either of these solutions or ask for more implementation details

Reply:
with the refactoring of Tree to a separate component the updates is done now

Reply:
although now, with the refactoring of Tree to a separate component the update is functioning.